gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/WriteChannelConfiguration.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.bigquery.model.JobConfigurationLoad;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.bigquery.JobInfo.CreateDisposition;
+import com.google.gcloud.bigquery.JobInfo.WriteDisposition;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Google BigQuery Configuration for a load operation. A load configuration can be used to load data
+ * into a table with a {@link com.google.gcloud.WriteChannel}
+ * ({@link BigQuery#writer(WriteChannelConfiguration)}).
+ */
+public class WriteChannelConfiguration implements LoadConfiguration, Serializable {
+
+  private static final long serialVersionUID = 470267591917413578L;
+
+  protected final TableId destinationTable;
+  protected final CreateDisposition createDisposition;
+  protected final WriteDisposition writeDisposition;
+  protected final FormatOptions formatOptions;
+  protected final Integer maxBadRecords;
+  protected final Schema schema;
+  protected final Boolean ignoreUnknownValues;
+  protected final List<String> projectionFields;
+
+  public static final class Builder implements LoadConfiguration.Builder {
+
+    private TableId destinationTable;
+    private CreateDisposition createDisposition;
+    private WriteDisposition writeDisposition;
+    private FormatOptions formatOptions;
+    private Integer maxBadRecords;
+    private Schema schema;
+    private Boolean ignoreUnknownValues;
+    private List<String> projectionFields;
+
+    protected Builder() {}
```

Comment:
Package scope?  (I could be missing something obvious with all my access-related comments/questions; let me know if so)